### DISCUSSION TITLE
How to avoid the distortion of a button?

### DISCUSSION THREAD

**Discussion by wsw70** [quasarframework/quasar#8970] [General - Components / Directives / etc] [answered]
I wanted to add a round button to a row of components:

```
<template>
  <div class="row q-my-md">
    <q-btn round color="secondary" icon="autorenew" @click="resetSelectedTags" class="q-mx-md"></q-btn>
    <fieldset class="row q-mx-md">
      <legend>Selected</legend>
      <q-chip
        v-for="(state, tag) in selectedTags"
        v-if="state"
        :key="tag"
        :label="tag"
        :style="{ background: tagToColor(tag) }"
        class="q-my-sm tag text-white"
        @click="updateTagState(tag, !state)"
        clickable
      />
...
```

The `q-btn` ended up distorted as if its height was driven by the height of the line:

![image](https://user-images.githubusercontent.com/1345886/115413990-f1293080-a1f5-11eb-862f-a484d81b1672.png)

I get the same effect with a standard (non-round) button:

![image](https://user-images.githubusercontent.com/1345886/115415785-7a8d3280-a1f7-11eb-8465-d23cfd7edb46.png)

**How can I avoid this?**

**Comment by metalsadman** [comment, marked as answer]
Wrap it in a div.

**Reply by wsw70** [reply]
Thank you, they are now of the proper size (and aligned to the top of the line, I will see how to fix this as it must be a `flex` behaviour)

**Reply by rstoenescu** [reply]
It is indeed a flex behavior. Alternatively you could also set a vertical aligment (`row items-center`).

**Reply by wsw70** [reply]
@rstoenescu Thanks. I posted another solution that gives the same quasi-centered alignment (without the `div` but with an extra class)

**Comment by wsw70** [comment]
After analyzing @metalsadman answer, I realized that the `div` is not a `flex` element and therefore is aligned to the top of the line.

The alternate solution I found is

    <q-btn round color="secondary" icon="autorenew" @click="resetSelectedTags" class="q-mx-md flex"></q-btn>

The `flex` class is

```
  .flex {
    display: flex;
    align-self: center;
  }
```

The alignment is not perfect but acceptable:

![image](https://user-images.githubusercontent.com/1345886/115440506-16c33380-a210-11eb-9639-a4f1e6aac106.png)
